lib/src/main/scala/code/chymyst/jc/Molecules.scala

```diff
@@ -146,20 +146,20 @@ private[jc] trait NonblockingMolecule[T] extends Molecule {
         v <- moleculeValues.getOne(this)
       } yield {
         usedInputs += (this -> v)
-        v.asInstanceOf[AbsMolValue[T]].getValue
+        v.asInstanceOf[AbsMolValue[T]].getValue // need the type cast due to erasure of type T
```

Comment:
see tksfz at http://stackoverflow.com/questions/1094173/how-do-i-get-around-type-erasure-on-scala-or-why-cant-i-get-the-type-paramete